applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -251,46 +253,50 @@ void DistanceModificationProcess::DeactivateFullNegativeElements() {
     ModelPart::NodesContainerType& rNodes = mrModelPart.Nodes();
     ModelPart::ElementsContainerType& rElements = mrModelPart.Elements();
 
-    // Initialize the ACTIVE flag to false
+    // Initialize the EMBEDDED_IS_ACTIVE variable flag to 0
     #pragma omp parallel for
-    for (int i_node = 0; i_node < static_cast<int>(rNodes.size()); ++i_node) {
+    for (int i_node = 0; i_node < static_cast<int>(rNodes.size()); ++i_node){
         ModelPart::NodesContainerType::iterator it_node = rNodes.begin() + i_node;
-        it_node->Set(ACTIVE, false);
+        if (it_node->FastGetSolutionStepValue(DISTANCE) < 0.0){
+            it_node->SetValue(EMBEDDED_IS_ACTIVE, 0);
+        }
     }
 
-    // Deactivate those elements whose fixed nodes and negative distance nodes summation is equal (or larger) to their number of nodes
+    // Deactivate those elements whose negative distance nodes summation is equal to their number of nodes
     #pragma omp parallel for
-    for (int k = 0; k < static_cast<int>(rElements.size()); ++k) {
-        unsigned int fixed = 0;
-        unsigned int inside = 0;
+    for (int k = 0; k < static_cast<int>(rElements.size()); ++k){
+        unsigned int n_neg = 0;
         ModelPart::ElementsContainerType::iterator itElement = rElements.begin() + k;
         GeometryType& rGeometry = itElement->GetGeometry();
 
         // Check the distance function sign at the element nodes
-        for (unsigned int i_node=0; i_node<rGeometry.size(); i_node++) {
-            if (rGeometry[i_node].FastGetSolutionStepValue(DISTANCE) < 0.0)
-                inside++;
-            if (rGeometry[i_node].IsFixed(VELOCITY_X) && rGeometry[i_node].IsFixed(VELOCITY_Y) && rGeometry[i_node].IsFixed(VELOCITY_Z))
-                fixed++;
+        for (unsigned int i_node=0; i_node<rGeometry.size(); i_node++){
+            if (rGeometry[i_node].FastGetSolutionStepValue(DISTANCE) < 0.0){
+                n_neg++;
+            }
         }
 
-        (inside+fixed >= rGeometry.size()) ? itElement->Set(ACTIVE, false) : itElement->Set(ACTIVE, true);
+        (n_neg == rGeometry.size()) ? itElement->Set(ACTIVE, false) : itElement->Set(ACTIVE, true);
 
         // If the element is ACTIVE, all its nodes are active as well
-        if (itElement->Is(ACTIVE)) {
-            for (unsigned int i_node = 0; i_node < rGeometry.size(); ++i_node) {
-                rGeometry[i_node].Set(ACTIVE, true);
+        if (itElement->Is(ACTIVE)){
+            for (unsigned int i_node = 0; i_node < rGeometry.size(); ++i_node){
+                rGeometry[i_node].SetLock();
+                int& activation_index = rGeometry[i_node].GetValue(EMBEDDED_IS_ACTIVE);
+                activation_index += 1;
```

Comment:
since you are syncing one single variable, here it would be back to use an atomic